data_model/src/query/mod.rs

```diff
@@ -62,7 +62,7 @@ macro_rules! queries {
 /// Trait for typesafe query output
 pub trait Query: Into<QueryBox> + seal::Sealed {
     /// Output type of query
-    type Output: Into<Value> + TryFrom<Value>;
+    type Output: Into<Value> + TryFrom<Value> + Encode;
```

Comment:
```suggestion
    type Output: Encode;
```

Do we still need these bounds?

Reply:
`TryFrom<Value>` is still needed for smart contracts AFAIU, but `Into<Value>` is not necessary, yeah

Reply:
what is it used for in smart contracts?

Reply:
what is it used for in smart contracts?

Reply:
`host_execute_query`'s result is decoded as `Result<Value, ValidationFail>`, which later needs a conversion into a concrete type https://github.com/hyperledger/iroha/blob/fcd2c28b7f104e95b0e1c228ba39e459c4b1083a/smart_contract/src/lib.rs#L101

Reply:
I think this is only a temporary limitation. It happens becaue we upper cast query into `QueryBox`. I'll split this into a separate ticket